NitroxTest/Server/EventTriggererTest.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections.Generic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NitroxModel.Utility;
+using NitroxServer.GameLogic;
+
+namespace NitroxTest.Server
+{
+    [TestClass]
+    public class EventTriggererTest
+    {
+        private EventTriggerer newEventTriggerer(double? elapsedTime) => new(null, elapsedTime.GetValueOrDefault(), null);
+        private EventTriggerer newEventTriggerer() => newEventTriggerer(null);
+
+        [TestMethod]
+        public void VerifyEventsOrder()
+        {
+            EventTriggerer eventTriggerer = newEventTriggerer();
+            List<string> eventsOrder = new() { "Story_AuroraWarning1", "Story_AuroraWarning2", "Story_AuroraWarning3", "Story_AuroraWarning4", "Story_AuroraExplosion" };
+            List<string> eventTriggererEvents = new(eventTriggerer.eventTimers.Keys);
+
+            bool goodOrder = true;
+            for (int i = 0; i < eventsOrder.Count; i++)
+            {
+                if (!eventsOrder[i].Equals(eventTriggererEvents[i]))
+                {
+                    goodOrder = false;
+                }
+            }
+
+            Assert.IsTrue(goodOrder);
+        }
+
+        [TestMethod]
+        public void AuroraExplosionAndWarningTime()
+        {
+            EventTriggerer eventTriggerer = newEventTriggerer();
+            Assert.AreEqual(eventTriggerer.eventTimers["Story_AuroraWarning4"].Interval, eventTriggerer.eventTimers["Story_AuroraExplosion"].Interval);
+        }
+
+        [TestMethod]
+        public void RealElapsedTimeMeasureUnit()
+        {
+            EventTriggerer eventTriggerer = newEventTriggerer(20d);
+            eventTriggerer.stopWatch.amplifiedElapsedMilliseconds = 500L;
+            Assert.AreEqual(eventTriggerer.stopWatch.ElapsedMilliseconds * 0.001 + eventTriggerer.ElapsedTime, eventTriggerer.GetRealElapsedTime());
+        }
+
+        [TestMethod]
+        public void TestSpeedChanges()
+        {
+            double initialInterval = 100000d;
+            long elapsedTime = 50000L;
+            AdjustableTimer timer = new() {
+                Interval = initialInterval,
+                Enabled = true,
+                AutoReset = false
+            };
+            timer.SetSpeed(5, elapsedTime);
+            elapsedTime += 10000L; // 2 seconds at speed x5
+            timer.SetSpeed(1, elapsedTime);
+            elapsedTime += 5000L; // 5 seconds at speed x1
+            timer.SetSpeed(3, elapsedTime);
+            elapsedTime += 15000L; // 5 seconds at speed x3
+            timer.SetSpeed(2, elapsedTime);
+            Assert.AreEqual((initialInterval - elapsedTime) / 2, timer.Interval); // divide by 2 because current speed would be x2
```

Comment:
Usually, when dealing with computing with big numbers and/or decimals compty, we don't use the strict equality == to validate that numbers are correct, It's better to introduce a delta (delta close to 0) check (i.e : Abs(new - old) < delta)